presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloSplitManager.java

```diff
@@ -0,0 +1,172 @@
+     *
+     * @param connectorId Connector ID
+     * @param client Client object for retrieving table metadata and splits
+     */
+    @Inject
+    public AccumuloSplitManager(AccumuloConnectorId connectorId, AccumuloClient client)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+        this.client = requireNonNull(client, "client is null");
+    }
+
+    /**
+     * Gets the source of splits from the given Presto table layout
+     *
+     * @param session Current client session
+     * @param layout Table layout
+     * @return Split source to pass splits to Presto for scan
+     */
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorSession session,
+            ConnectorTableLayoutHandle layout)
+    {
+        AccumuloTableLayoutHandle layoutHandle =
+                checkType(layout, AccumuloTableLayoutHandle.class, "layout");
+        AccumuloTableHandle tableHandle = layoutHandle.getTable();
+
+        String schemaName = tableHandle.getSchema();
+        String tableName = tableHandle.getTable();
+        String rowIdName = tableHandle.getRowId();
+
+        // If the column filter optimization is enabled, then get the list of column constraints to
+        // pack into a split
+        // These constraints are later converted to an Accumulo iterator and run on the tablet
+        // server
+        List<AccumuloColumnConstraint> iteratorConstraints;
+        if (AccumuloSessionProperties.isOptimizeColumnFiltersEnabled(session)) {
+            iteratorConstraints = getColumnConstraints(rowIdName, layoutHandle.getConstraint());
+        }
+        else {
+            iteratorConstraints = ImmutableList.of();
+        }
+
+        // Get the row domain column range
+        Domain rDom = getRangeDomain(rowIdName, layoutHandle.getConstraint());
+
+        // Call out to our client to retrieve all tablet split metadata using the row ID domain
+        // and the secondary index, if enabled and proper to do so
+        List<TabletSplitMetadata> tSplits = client.getTabletSplits(session, schemaName, tableName,
+                rDom, getColumnConstraints(rowIdName, layoutHandle.getConstraint()),
+                tableHandle.getSerializerInstance());
+
+        // Pack the tablet split metadata into a connector split
+        List<ConnectorSplit> cSplits = new ArrayList<>();
+        for (TabletSplitMetadata smd : tSplits) {
+            AccumuloSplit split = new AccumuloSplit(connectorId, schemaName, tableName, rowIdName,
+                    tableHandle.getSerializerClassName(), smd.getRanges(), iteratorConstraints,
+                    tableHandle.getScanAuthorizations(), smd.getHostPort());
+            LOG.debug("Added split %s", split);
+            cSplits.add(split);
+        }
+
+        Collections.shuffle(cSplits);
+
+        // TODO Would this be more beneficial to return splits in batches?
```

Comment:
I mean if it'd be better to implement a `SplitSource` that is asynchronous, feeding `AccumuloSplits` as they are being generated instead of one giant batch.  I expect it would be more efficient; I am unsure if it is worth the effort of implementation.